pkg/gen/input/makefile/internal/file/makefile.go

```diff
@@ -39,6 +52,33 @@ build-darwin: $(APPLICATION)-darwin
 ## build-linux: builds a local binary for linux/amd64
 build-linux: $(APPLICATION)-linux
 
+{{- if eq .CurrentFlavour .FlavourCLI}}
+
+.PHONY: package-darwin package-linux
+## package-darwin: prepares a packaged darwin/amd64 version
+package-darwin: $(APPLICATION)-package-darwin
+## package-linux: prepares a packaged linux/amd64 version
+package-linux: $(APPLICATION)-package-linux
+
+$(APPLICATION)-package-%: $(SOURCES)
+	@$(MAKE) $(APPLICATION)-$*
+	@$(MAKE) $(APPLICATION)-rename-binary-$*
+	@$(MAKE) $(APPLICATION)-archive-$*
+
+$(APPLICATION)-rename-binary-%:
+	cp $(APPLICATION)-$* $(APPLICATION)-$(VERSION)-$*-amd64
+
+$(APPLICATION)-archive-%:
+	mkdir -p $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-$*-amd64
+	mv $(APPLICATION)-$(VERSION)-$*-amd64 $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-$*-amd64
+	cp ./{README.md,LICENSE} $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-$*-amd64
+	cd $(PACKAGE_DIR)/ && \
+	tar -cvzf $(APPLICATION)-$(VERSION)-$*-amd64.tar.gz \
+		$(APPLICATION)-$(VERSION)-$*-amd64
+	rm -rf $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-$*-amd64
```

Comment:
I think we should rely more on dependencies here. I made this work:

```
.PHONY: build build-darwin build-linux
## build: builds a local binary
build: $(APPLICATION)
## build-darwin: builds a local binary for darwin/amd64
build-darwin: $(APPLICATION)-darwin
## build-linux: builds a local binary for linux/amd64
build-linux: $(APPLICATION)-linux

$(APPLICATION): $(APPLICATION)-$(VERSION)-$(OS)-amd64
	cp -a $< $@

$(APPLICATION)-%: $(APPLICATION)-$(VERSION)-%-amd64
	cp -a $< $@

$(APPLICATION)-$(VERSION)-%-amd64: $(SOURCES)
	GOOS=$* GOARCH=amd64 go build -ldflags "$(LDFLAGS)" -o $@ .

.PHONY: package-darwin package-linux
## package-darwin: prepares a packaged darwin/amd64 version
package-darwin: $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-darwin-amd64.tar.gz
## package-linux: prepares a packaged linux/amd64 version
package-linux: $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-linux-amd64.tar.gz

$(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-%-amd64.tar.gz: DIR=$(PACKAGE_DIR)/$<
$(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-%-amd64.tar.gz: $(APPLICATION)-$(VERSION)-%-amd64
	mkdir -p $(DIR)
	cp $< $(DIR)
	cp README.md LICENSE $(DIR)
	tar -C $(PACKAGE_DIR) -cvzf $<.tar.gz $<
	rm -rf $(DIR)
```

Reply:
Output looks like:

```
$ make package-darwin
mkdir -p ./bin-dist/devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
cp devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64 ./bin-dist/devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
cp README.md LICENSE ./bin-dist/devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
tar -C ./bin-dist -cvzf devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64.tar.gz devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
a devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
a devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64/LICENSE
a devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64/devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
a devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64/README.md
rm -rf ./bin-dist/devctl-0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-darwin-amd64
```

Reply:
Does `package-linux` work for you in that case?
For me, `package-darwin` works great, but `package-linux` uses `GOOS` equal to `bin-dist/0.0.0-65a8a4f91df2595110daca6968b3db8d552d3d49-linux-amd64.tar.gz`, for some reason.

Reply:
Same for me. Man this is magical. Let's go with that. You need to deliver. We can fix that later if target names don't change.

Reply:
Alright. I at least understand why this _can_ be https://www.gnu.org/software/make/manual/html_node/Pattern-Match.html

Reply:
I found it:

```
.PHONY: build build-darwin build-linux
## build: builds a local binary
build: $(APPLICATION)
## build-darwin: builds a local binary for darwin/amd64
build-darwin: $(APPLICATION)-darwin
## build-linux: builds a local binary for linux/amd64
build-linux: $(APPLICATION)-linux

$(APPLICATION): $(APPLICATION)-$(VERSION)-$(OS)-amd64
	cp -a $< $@

$(APPLICATION)-darwin: $(APPLICATION)-$(VERSION)-darwin-amd64
	cp -a $< $@

$(APPLICATION)-linux: $(APPLICATION)-$(VERSION)-linux-amd64
	cp -a $< $@

$(APPLICATION)-$(VERSION)-%-amd64: $(SOURCE)
	GOOS=$* GOARCH=amd64 go build -ldflags "$(LDFLAGS)" -o $@ .

.PHONY: package-darwin package-linux
## package-darwin: prepares a packaged darwin/amd64 version
package-darwin: $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-darwin-amd64.tar.gz
## package-linux: prepares a packaged linux/amd64 version
package-linux: $(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-linux-amd64.tar.gz

$(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-%-amd64.tar.gz: DIR=$(PACKAGE_DIR)/$<
$(PACKAGE_DIR)/$(APPLICATION)-$(VERSION)-%-amd64.tar.gz: $(APPLICATION)-$(VERSION)-%-amd64
	mkdir -p $(DIR)
	cp $< $(DIR)
	cp README.md LICENSE $(DIR)
	tar -C $(PACKAGE_DIR) -cvzf $<.tar.gz $<
	rm -rf $(DIR)
```

Replaced `$(APPLICATION)-%` with `$(APPLICATION)-darwin` and `$(APPLICATION)-linux`.

Reply:
That's awesome, I'll make the changes in a separate PR